app/main/views.py

```diff
@@ -517,3 +520,116 @@ def mousereport_pdf(animal_id):
     stylesheets = [CSS(url_for('static', filename='styles.css')),
                    CSS(url_for('static', filename='datajoint.css'))]
     return render_pdf(HTML(string=html), stylesheets=stylesheets)
+
+
+@main.route('/surgery', methods=['GET', 'POST'])
+def surgery():
+    fexperiment = dj.create_virtual_module("csmith_testing", "csmith_testing")
+    form = forms.SurgeryForm(request.form)
+    if request.method == 'POST' and form.validate():
+        tuple_ = {'animal_id': form['animal_id'].data, 'date': str(form['date'].data),
+                  'time': str(form['time_input'].data), 'username': form['user'].data,
+                  'surgery_outcome': form['outcome'].data, 'surgery_quality': form['surgery_quality'].data,
+                  'surgery_type': form['surgery_type'].data, 'weight': form['weight'].data,
+                  'ketoprofen': form['ketoprofen'].data, 'notes': form['notes'].data}
+        status_tuple_ = {'animal_id': tuple_['animal_id'], 'date': tuple_['date'], 'checkup_notes': ''}
+        if tuple_['ketoprofen'] == 0:
+            tuple_.pop('ketoprofen')
+        if tuple_['weight'] == 0:
+            tuple_.pop('weight')
+        if not fexperiment.Surgery().proj() & tuple_:
+            try:
+                fexperiment.Surgery().insert1(tuple_)
+                fexperiment.SurgeryStatus.insert1(status_tuple_)
+                flash('Inserted record for animal {}'.format(tuple_['animal_id']))
+            except Exception as ex:
+                ex_message = "An exception of type {} occurred. More information:\n\n{}".format(type(ex).__name__,
+                                                                                                ex.args)
+                flash(ex_message)
+        else:
+            flash('Record already exists.')
+    return render_template('surgery.html', form=form)
+
+
+@main.route('/surgery/status', methods=['GET', 'POST'])
+def surgery_status():
+    fexperiment = dj.create_virtual_module("csmith_testing", "csmith_testing")
+    form = forms.SurgeryStatusForm(request.form)
+    surgery_data = (fexperiment.Surgery & {'surgery_outcome': 'Survival'}).fetch(as_dict=True)
+    newest_status = []
+    for entry in surgery_data:
```

Comment:
`for status_key in (fexperiment.Surgery & {'surgery_outcome': 'Survival'}).proj():` will return status_keys (I think).  If not, `for status_key in (fexperiment.Surgery & {'surgery_outcome': 'Survival'}).fetch('KEY'):` will do for sure